src/classes/RD2_RecurringDonations_TEST.cls

```diff
@@ -173,31 +173,121 @@ private with sharing class RD2_RecurringDonations_TEST {
-                        .withAmount(10)
-                        .withInstallmentPeriodMonthly()
-                        .withInstallmentFrequency(1)
-                        .withDayOfMonth('20')
-                        .withDateEstablished(Date.newInstance(2019, 9, 15))
-                        .withStartDate(Date.newInstance(2019, 9, 15))
-                        .build()
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withContact(contact.Id)
+                .withAmount(10)
+                .withInstallmentPeriodMonthly()
+                .withInstallmentFrequency(1)
+                .withDayOfMonth('20')
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2019, 9, 15))
+                .build()
         };
 
         Test.startTest();
         insert rds;
         Test.stopTest();
 
-        Map<Id, npe03__Recurring_Donation__c> actualRdById = new Map<Id, npe03__Recurring_Donation__c>(
-                rdGateway.getRecords(rds)
-        );
+        Map<Id, npe03__Recurring_Donation__c> actualRdById = new Map<Id, npe03__Recurring_Donation__c>(rdGateway.getRecords(rds));
         System.assertEquals(rds.size(), actualRdById.size(), 'Number of RDs should match');
 
         List<RecurringDonationSchedule__c> rdSchedules = rdScheduleGateway.getRecords(rds);
         System.assertEquals(rds.size(), rdSchedules.size(), 'Number of RD schedule records should equal the number of RD records');
 
     }
 
+    /***
+    * @description Verifies new schedule is created when recurring donation is updated
+    */
+    @isTest
+    private static void shouldCreateNewRDScheduleOnUpdate() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact contact = getContact();
+
+        List<npe03__Recurring_Donation__c> rds = new List<npe03__Recurring_Donation__c>{
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withContact(contact.Id)
+                .withAmount(10)
+                .withInstallmentPeriodMonthly()
+                .withInstallmentFrequency(1)
+                .withDayOfMonth('20')
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2019, 9, 15))
+                .build()
+        };
+
+        Test.startTest();
+        insert rds;
+        rds[0].InstallmentFrequency__c = 3;
+        update rds;
+        Test.stopTest();
+
+        Map<Id, npe03__Recurring_Donation__c> actualRdById = new Map<Id, npe03__Recurring_Donation__c>(rdGateway.getRecords(rds));
+        System.assertEquals(rds.size(), actualRdById.size(), 'Number of RDs should match');
+
+        List<RecurringDonationSchedule__c> rdSchedules = rdScheduleGateway.getRecords(rds);
+        System.assertEquals(rds.size()+1, rdSchedules.size(), 'Number of RD schedule records should be one greater than the number of RD records');
```

Comment:
Would be good if this verified the schedule record changes in additional to fact that a new schedule record was created.